exporter/clickhouseexporter/internal/summary_metrics.go

```diff
@@ -0,0 +1,155 @@
+SETTINGS index_granularity=8192, ttl_only_drop_parts = 1;
+`
+	// language=ClickHouse SQL
+	insertSummaryTableSQL = `INSERT INTO %s_summary (
+	ResourceAttributes,
+    ResourceSchemaUrl,
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+    ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+	StartTimeUnix,
+	TimeUnix,
+    Count,
+    Sum,
+    ValueAtQuantiles.Quantile,
+	ValueAtQuantiles.Value,
+    Flags) VALUES `
+	summaryPlaceholders = "(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)"
+)
+
+type summaryModel struct {
+	metricName        string
+	metricDescription string
+	metricUnit        string
+	summary           pmetric.Summary
+}
+
+type summaryMetrics struct {
+	summaryModel []*summaryModel
+	metadata     *MetricsMetaData
+	insertSQL    string
+}
+
+func (s *summaryMetrics) insert(ctx context.Context, tx *sql.Tx, logger *zap.Logger) error {
+	var valuePlaceholders []string
+	var valueArgs []interface{}
+
+	for _, model := range s.summaryModel {
+		for i := 0; i < model.summary.DataPoints().Len(); i++ {
+			dp := model.summary.DataPoints().At(i)
+			valuePlaceholders = append(valuePlaceholders, summaryPlaceholders)
+
+			valueArgs = append(valueArgs, s.metadata.ResAttr)
+			valueArgs = append(valueArgs, s.metadata.ResURL)
+			valueArgs = append(valueArgs, s.metadata.ScopeInstr.Name())
+			valueArgs = append(valueArgs, s.metadata.ScopeInstr.Version())
+			valueArgs = append(valueArgs, attributesToMap(s.metadata.ScopeInstr.Attributes()))
+			valueArgs = append(valueArgs, s.metadata.ScopeInstr.DroppedAttributesCount())
+			valueArgs = append(valueArgs, s.metadata.ScopeURL)
+			valueArgs = append(valueArgs, model.metricName)
+			valueArgs = append(valueArgs, model.metricDescription)
+			valueArgs = append(valueArgs, model.metricUnit)
+			valueArgs = append(valueArgs, attributesToMap(dp.Attributes()))
+			valueArgs = append(valueArgs, dp.StartTimestamp().AsTime().UnixNano())
+			valueArgs = append(valueArgs, dp.Timestamp().AsTime().UnixNano())
+			valueArgs = append(valueArgs, dp.Count())
+			valueArgs = append(valueArgs, dp.Sum())
+
+			quantiles, values := convertValueAtQuantile(dp.QuantileValues())
+			valueArgs = append(valueArgs, quantiles)
+			valueArgs = append(valueArgs, values)
+			valueArgs = append(valueArgs, uint32(dp.Flags()))
+		}
+	}
+
+	if len(valuePlaceholders) == 0 {
+		return nil
+	}
+
+	start := time.Now()
+	_, err := tx.ExecContext(ctx, fmt.Sprintf("%s %s", s.insertSQL, strings.Join(valuePlaceholders, ",")), valueArgs...)
+	if err != nil {
+		return fmt.Errorf("insert summary metrics fail:%w", err)
+	}
+	duration := time.Since(start)
+
+	// TODO latency metrics
+	logger.Debug("insert summary metrics", zap.Int("records", len(valuePlaceholders)),
+		zap.String("cost", duration.String()))
+	return nil
+}
+
+func (s *summaryMetrics) Add(metrics interface{}, name string, description string, unit string) {
+	summary, _ := metrics.(pmetric.Summary)
```

Comment:
should check the conversion to avoid panics, and return an error

Reply:
done